### DISCUSSION TITLE
Can I use a markdown template (not HTML template) and populate contents for different markdown instances?

### DISCUSSION THREAD

**Discussion by GarfieldJiang** [mkdocs/mkdocs#3755] [Q&A]
I know the markdown-include plugin, but it after all is not the ideal way to do what I want.

For example. MyTemplate.md contains an outline of a doc. Instances A.md, B.md, C.md want to use the outline and populate things as params into it.

**Comment by kamilkrzyskow** [comment]
Hello @GarfieldJiang,
you probably want the `mkdocs-macros-plugin` https://mkdocs-macros-plugin.readthedocs.io/en/stable/
There is also an `include` expression/function, and you can add global (mkdocs.yml) and local (parent view that includes the fragment) variables that will be used inside the imported file.
